homeassistant/config_entries.py

```diff
@@ -1260,13 +1260,25 @@ async def async_init(
         if not context or "source" not in context:
             raise KeyError("Context not set or doesn't have a source set")
 
+        # reauth/reconfigure flows should be linked to a config entry
+        if (source := context["source"]) in {
+            SOURCE_REAUTH,
+            SOURCE_RECONFIGURE,
+        } and "entry_id" not in context:
+            # Deprecated in 2024.12, should fail in 2025.12
+            report(
+                f"initialises a {source} flow for integration {handler} without "
+                "a link to the config entry. This will stop working in 2025.12",
+                error_if_integration=False,
+                error_if_core=False,
```

Comment:
Do we need to log for core integrations? Can't we fix all those places now already if there are any?

Reply:
I looked through the code to try and find them - fixed a few but I'm worried about having missed some.

Reply:
There are only 86 results for `flow.async_init` when excluding tests. Should be ok to check, I think. The ones in the tests should fail immediately.

Reply:
I checked the 86 results and didn't find any with the relevant sources.

Reply:
I marked as draft as I assume the tests will need adjusting... one for tomorrow now 🌙